Src/FluentAssertions/Specialized/FunctionAssertions.cs

```diff
@@ -11,8 +12,11 @@ namespace FluentAssertions.Specialized
     [DebuggerNonUserCode]
     public class FunctionAssertions<T> : DelegateAssertions<Func<T>>
     {
-        public FunctionAssertions(Func<T> subject, IExtractExceptions extractor) : base(subject, extractor)
+        private readonly IClock clock;
+
+        public FunctionAssertions(Func<T> subject, IExtractExceptions extractor, IClock clock) : base(subject, extractor, clock)
```

Comment:
🔧 This and other classes changes a public constructor to take an additional mandatory `IClock`.
We could keep a constructor without an `IClock` and default to `Clock`

```c#
public FunctionAssertions(Func<T> subject, IExtractExceptions extractor)
    : this(subject, extractor, new Clock())
{}
```

Reply:
Good catch and great idea.